llvm/lib/CheerpUtils/TypeOptimizer.cpp

```diff
@@ -446,7 +448,9 @@ TypeOptimizer::TypeMappingInfo TypeOptimizer::rewriteType(Type* t)
 		}
 
 		// Generate a new type inconditionally, it may end up being the same as the old one
```

Comment:
here the comment is incorrect now. we generate a new type except when the old one is a literal. in that case, at the end we will "get" the (possibly the same) new literal type.

Reply:
Reworded the comment so the new behaviour is more clear